correlation-parser/src/timer.rs

```diff
@@ -37,16 +37,15 @@ impl Watchdog {
                         Err(TryRecvError::Empty) => (),
                     }
                 } else {
-                    thread::sleep(delta);
-
                     match rx.try_recv() {
                         Ok(ControlEvent::Stop) | Err(TryRecvError::Disconnected) => break,
                         Ok(ControlEvent::Park) => { is_parking = true; }
                         Ok(ControlEvent::UnPark) => { is_parking = false; }
-                        Err(TryRecvError::Empty) => (),
+                        Err(TryRecvError::Empty) => {
+                            cb();
+                            thread::sleep(delta);
+                        }
```

Comment:
But callback was called in every cases independently from the state of the is_parking. Now, it will be called only if is_parking was false. Is it the correct behavior?